packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -58,7 +58,7 @@ export async function load_node({
 				/** @type {import('types').SSRPage} */ (route),
 				event,
 				options,
-				!!state.prerender
+				node.module.prerender ?? state?.prerender?.default ?? false
```

Comment:
It took me awhile to read this line. It might help to refactor it out into a variable since the variable name would make things clearer
```
const prerender_enabled = node.module.prerender ?? state?.prerender?.default ?? false;
```

Do we still need the `state.prerender` reference? I'd expect it to be something like `prerender_enabled && state.prerender`

Reply:
Agreed, a variable makes that easier to understand. Added.

> Do we still need the state.prerender reference? I'd expect it to be something like prerender_enabled && state.prerender

I'm not sure _exactly_ what you mean there. We have to check for `node.module.prerender`, as that corresponds to the module-scoped `export const prerender`. If that's not defined, we have to check for the default behavior (if `kit.prerender.default === true`, we should prerender this page _unless_ the page opts out, which we already checked for).

Reply:
I could be wrong about that. I haven't really looked at any of the shadow endpoint code yet. I'll let Rich take a look at this PR and the remaining questions as he's more familiar

Reply:
Yeah, full disclosure, I _definitely_ don't understand everything in the codebase, so I could definitely be wrong here!